### PR TITLE
ANTHA-2518: Empty return if units is missing.

### PR DESCRIPTION
**Pull request by cosmonautspiff**
Instead of triggering a nil pointer error, the methods will return an empty variable.

Please check the differences and feel free to change how this was done.

Beware, that there will be **no** error anymore if someone forgets to enter units. Is that preferable? Perhaps we should just implement a UI check.

### CODE REVIEW

**Comment by x-anderson** [comment]
Looks good and really useful for catching these errors here, instead of elements! 👍  The Travis failing - is this due the the Errorf call without an argument?

File: antha/anthalib/wunit/unit.go [hunk 1]
```diff
@@ -140,16 +153,25 @@ func (self *Unit) PrefixedSymbol() string {
...
+	if self == nil {
+		return ""
+	}
 	return self.siSymbol
 }
 
 // getConversionFactor get the conversion factor between this unit and rhs.
 func (self *Unit) getConversionFactor(rhs *Unit) (float64, error) {
+	if self == nil || rhs == nil {
+		return 0.0, errors.Errorf("cannot convert units: nil units provided")
```
Comment (x-anderson) [inline antha/anthalib/wunit/unit.go]: Errorf should require an argument passed in for formatting.

If both self and rhs are nil - should a nil error be returned anyway?

File: antha/anthalib/wunit/unit.go [hunk 2]
```diff
@@ -109,21 +109,34 @@ func (self *Unit) GobDecode(b []byte) error {
...
+		return ""
+	}
 	return fmt.Sprintf("%s[%s]", self.Name(), self.PrefixedSymbol())
 }
 
 // Prefix get the SI prefix of this unit, or " " if none
 func (self *Unit) Prefix() SIPrefix {
+	if self == nil {
+    var emptyPrefix SIPrefix
+		return emptyPrefix
```
Comment (x-anderson) [inline antha/anthalib/wunit/unit.go]: Could save a line here and write:

```
return SIPrefix{}
```

and remove

```
var emptyPrefix SIPrefix
```

File: antha/anthalib/wunit/unit.go (hunk 1 above)
Comment (cosmonautspiff) [reply antha/anthalib/wunit/unit.go]: Used errors.New

Sorry, not sure I understand the nil error comment

File: antha/anthalib/wunit/unit.go (hunk 2 above)
Comment (cosmonautspiff) [reply antha/anthalib/wunit/unit.go]: done

**Comment by cosmonautspiff** [comment]
My bad, I thought that building antha would format everything.

**Comment by cosmonautspiff** [comment]
If this is fine, please merge.

**Pull request merged**